compiler/rustc_monomorphize/src/partitioning/mod.rs

```diff
@@ -207,15 +207,19 @@ pub fn partition<'tcx>(
     if instrument_dead_code {
         // Find the smallest CGU that has exported symbols and put the dead
         // function stubs in that CGU. We look for exported symbols to increase
-        // the likelyhood the linker won't throw away the dead functions.
-        let mut cgus_with_exported_symbols: Vec<_> = post_inlining
-            .codegen_units
-            .iter_mut()
+        // the likelihood the linker won't throw away the dead functions.
+        // FIXME(#92165): In order to truly resolve this, we need to make sure
+        // the object file (CGU) containing the dead function stubs is included
+        // in the final binary. This will probably require forcing these
+        // function symbols to be included via `-u` or `/include` linker args.
+        let mut cgus: Vec<_> = post_inlining.codegen_units.iter_mut().collect();
+        cgus.sort_by_key(|cgu| cgu.size_estimate());
+
+        let dead_code_cgu = cgus
+            .into_iter()
             .filter(|cgu| cgu.items().iter().any(|(_, (linkage, _))| *linkage == Linkage::External))
-            .collect();
-        cgus_with_exported_symbols.sort_by_key(|cgu| cgu.size_estimate());
-
-        let dead_code_cgu = cgus_with_exported_symbols.last_mut().unwrap();
+            .last()
```

Comment:
This is still going through every CGU to check whether it contains symbols with external linkage. Can you reverse the iterator instead and then take `.next()`?

Reply:
Or just sort by `usize::MAX - cgu.size_estimate`, to avoid having to reverse the iterator.

Reply:
Done.